google-http-client-apache-v5/src/test/java/com/google/api/client/http/apache/v5/Apache5MockHttpResponse.java

```diff
@@ -0,0 +1,182 @@
+package com.google.api.client.http.apache.v5;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.HttpVersion;
+import org.apache.hc.core5.http.ProtocolException;
+import org.apache.hc.core5.http.ProtocolVersion;
+
+public class Apache5MockHttpResponse implements ClassicHttpResponse {
```

Comment:
I would be wary about making `Apache5MockHttpResponse` implement `ClassicHttpResponse` (the Apache class). This doesn't reflect the production class structure in which `Apache5HttpResponse extends LowLevelHttpResponse` (our abstraction class).

This could get confusing, as it will make the tests read differently than the production logic would look.

Reply:
The purpose of that class is to mock a ClassicHttpResponse produced by the underlying Apache 5 client indeed, not to mock the `LowLevelHttpResponse` created and managed by `Apache5HttpTransport`. For example, in this code section we set up the client to return such mock response but our `Apache5HttpRequest` object still obtains a `LowLevelHttpResponse`

https://github.com/googleapis/google-http-java-client/blob/871901c9c670771b1f599415d6989725f58a16c9/google-http-client-apache-v5/src/test/java/com/google/api/client/http/apache/v5/Apache5HttpTransportTest.java#L212-L230

I will rename it to reflect that it is intended to mock an Apache 5 HttpClient's response.

Reply:
Renamed
 * `Apache5MockHttpResponse` -> `MockClassicHttpResponse`
 * `Apache5MockHttpClient` -> `MockHttpClient`